### PR TITLE
feat: aside header top alert

### PR DESCRIPTION
**Pull request by goshander**
I want to implement something like this
image

### CODE REVIEW

File: src/components/AsideHeader/AsideHeader.tsx [hunk 1]
```diff
@@ -1,32 +1,59 @@
...
+        }
+        return () => window.removeEventListener('resize', updateTopSize);
+    }, [topAlert, topRef, topHeight]);
+
     return (
         <AsideHeaderContextProvider value={asideHeaderContextValue}>
             <AsideHeaderInnerContextProvider value={asideHeaderInnerContextValue}>
                 <div className={b({compact}, className)}>
+                    {/* Top Panel */}
+                    <TopPanel ref={topRef} />
```
Comment (DarkGenius) [inline src/components/AsideHeader/AsideHeader.tsx:49]: Can we not render the component `topAlert` is not set?

Comment (goshander) [reply src/components/AsideHeader/AsideHeader.tsx]: fixed

**Comment by goshander** [comment]
@DarkGenius i think need waiting this [PR #140](https://github.com/gravity-ui/navigation/pull/140), right?

**Comment by DarkGenius** [comment]
> @DarkGenius i think need waiting this [PR #140](https://github.com/gravity-ui/navigation/pull/140), right?

Yes, it’s possible that after this PR is merged, changes will be required in your PR

File: src/components/AsideHeader/types.tsx [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+    AsideHeaderTopAlertProps,
+} from '../types';
 import {AsideHeaderContextType} from './AsideHeaderContext';
 
 export interface LayoutProps {
     compact: boolean;
     className?: string;
+    topRef?: React.RefObject<HTMLDivElement>;
+    topAlert?: AsideHeaderTopAlertProps;
+    updateTopSize?: () => void;
```
Comment (DarkGenius) [inline src/components/AsideHeader/types.tsx:17]: It's better to rename it to `onCloseTopAlert`. Because arbitrary code can be passed to this property.

**Review by DarkGenius — COMMENTED** [review]
The top panel overlaps.
telegram-cloud-photo-size-2-5395780814719012458-y

File: src/components/AsideHeader/__stories__/AsideHeaderShowcase.scss [hunk 3]
```diff
@@ -80,7 +80,7 @@ body {
     &__settings-panel,
     &__search-panel {
         width: 300px;
-        height: 100%;
+        height: calc(100% - 40px);
```
Comment (goshander) [inline src/components/AsideHeader/__stories__/AsideHeaderShowcase.scss:83]: fix drawer padding at 20px for remove unexpected scroll

File: src/components/AsideHeader/useAsideHeaderTopPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    React.useLayoutEffect(() => {
+        const updateTopSizeDebounce = debounceFn(updateTopSize, 200, {leading: true});
+
+        if (topAlert) {
+            window.addEventListener('resize', updateTopSizeDebounce);
+            updateTopSizeDebounce();
+        }
+        return () => {
+            window.removeEventListener('resize', updateTopSizeDebounce);
+            setAsideTopPanelHeight(0);
```
Comment (goshander) [inline src/components/AsideHeader/useAsideHeaderTopPanel.tsx:52]: set top panel size to zero for correct destroy component behaviour with iframe switching at storybook

**Comment by goshander** [comment]
> The top panel overlaps. telegram-cloud-photo-size-2-5395780814719012458-y

fixed

File: src/components/AsideHeader/__stories__/AsideHeaderShowcase.tsx [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import React, {FC, useState} from 'react';
+import React, {FC} from 'react';
```
Comment (DarkGenius) [inline src/components/AsideHeader/__stories__/AsideHeaderShowcase.tsx:1]: Maybe then we’ll remove `FC` import too?

File: src/components/AsideHeader/AsideHeader.scss [hunk 6]
```diff
@@ -2,6 +2,10 @@
 
 $block: '.#{variables.$ns}aside-header';
 
+:root {
+    --gn-aside-top-panel-height: 0px;
```
[the remark this answers is not in the document]
Comment (goshander) [reply src/components/AsideHeader/AsideHeader.scss:6]: Then we can not change this value from document object

File: src/components/AsideHeader/__stories__/AsideHeaderShowcase.tsx (hunk 5 above)
Comment (goshander) [reply src/components/AsideHeader/__stories__/AsideHeaderShowcase.tsx:1]: fixed

File: src/components/AsideHeader/AsideHeader.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (DarkGenius) [reply src/components/AsideHeader/AsideHeader.scss:6]: We can use the correct selector

[the remark this answers is not in the document]
Comment (goshander) [reply src/components/AsideHeader/AsideHeader.scss]: > We can use the correct selector

Ok, changed to `getElementsByClassName` selector

**Review by DarkGenius — APPROVED** [review]

**Pull request merged**